src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIResponseChatClient.cs

```diff
@@ -0,0 +1,582 @@
+    public OpenAIResponseChatClient(OpenAIResponseClient responseClient)
+    {
+        _ = Throw.IfNull(responseClient);
+
+        _responseClient = responseClient;
+
+        // https://github.com/openai/openai-dotnet/issues/215
+        // The endpoint and model aren't currently exposed, so use reflection to get at them, temporarily. Once packages
+        // implement the abstractions directly rather than providing adapters on top of the public APIs,
+        // the package can provide such implementations separate from what's exposed in the public API.
+        Uri providerUrl = typeof(OpenAIResponseClient).GetField("_endpoint", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)
+            ?.GetValue(responseClient) as Uri ?? DefaultOpenAIEndpoint;
+        string? model = typeof(OpenAIResponseClient).GetField("_model", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)
+            ?.GetValue(responseClient) as string;
+
+        _metadata = new("openai", providerUrl, model);
+    }
+
+    /// <summary>Gets or sets <see cref="JsonSerializerOptions"/> to use for any serialization activities related to tool call arguments and results.</summary>
+    public JsonSerializerOptions ToolCallJsonSerializerOptions
+    {
+        get => _toolCallJsonSerializerOptions;
+        set => _toolCallJsonSerializerOptions = Throw.IfNull(value);
+    }
+
+    /// <inheritdoc />
+    object? IChatClient.GetService(Type serviceType, object? serviceKey)
+    {
+        _ = Throw.IfNull(serviceType);
+
+        return
+            serviceKey is not null ? null :
+            serviceType == typeof(ChatClientMetadata) ? _metadata :
+            serviceType == typeof(OpenAIResponseClient) ? _responseClient :
+            serviceType.IsInstanceOfType(this) ? this :
+            null;
+    }
+
+    /// <inheritdoc />
+    public async Task<ChatResponse> GetResponseAsync(
+        IEnumerable<ChatMessage> messages, ChatOptions? options = null, CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNull(messages);
+
+        // Convert the inputs into what OpenAIResponseClient expects.
+        var openAIResponseItems = ToOpenAIResponseItems(messages, ToolCallJsonSerializerOptions);
+        var openAIOptions = ToOpenAIResponseCreationOptions(options);
+
+        // Make the call to the OpenAIResponseClient.
+        var openAIResponse = (await _responseClient.CreateResponseAsync(openAIResponseItems, openAIOptions, cancellationToken).ConfigureAwait(false)).Value;
+
+        // Convert and return the results.
+        ChatResponse response = new()
+        {
+            ResponseId = openAIResponse.Id,
+            CreatedAt = openAIResponse.CreatedAt,
+            FinishReason = ToFinishReason(openAIResponse.IncompleteStatusDetails?.Reason),
+            Messages = [new(ChatRole.Assistant, (string?)null)],
```

Comment:
Minor detail, and maybe you have some reason not to do this:

```suggestion
            Messages = [new(ChatRole.Assistant, [])],
```

I found the `(string?)null` approach a bit confusing as it seems like it's creating an empty `TextContent` which would be undesirable, but then I realised the `ChatMessage` constructor special-cases `null` and omits the content. It's a bit difficult to realise that's going to happen so may be clearer to pass `[]`, unless there's some other reason I didn't notice.